corehq/apps/linked_domain/views.py

```diff
@@ -446,6 +440,27 @@ def create_remote_report_link(self, in_data):
             return {'success': False}
 
 
+def handle_create_domain_link_request(couch_user, upstream_domain, downstream_domain):
+    if not domain_exists(downstream_domain):
+        return ugettext("The project space {} does not exist. Make sure the name is correct and that "
+                        "this domain hasn't been deleted.").format(downstream_domain)
+
+    if is_active_link(upstream_domain, downstream_domain):
+        return ugettext(
+            "The project space {} is already a downstream project space of {}."
+        ).format(downstream_domain, upstream_domain)
+
+    if not user_has_admin_access_in_all_domains(couch_user, [upstream_domain, downstream_domain]):
+        return ugettext("The user must be an admin is both project spaces to successfully create a link.")
+
+    try:
+        DomainLink.link_domains(downstream_domain, upstream_domain)
+    except DomainLinkError:
+        return ugettext(
+            "An error was encountered while attempting to link {} to {}."
```

Comment:
This is obscuring the more detailed error message that `link_domains` throws, it ought to show that error instead.

Reply:
Yeah I had mixed feelings here, and have maybe brought this up before. Should we be returning the actual exception to the UI? That doesn't feel great to me, but also understand if we choose to return a simplified/generic message, we should track the actual exception in sentry.

Reply:
A lot of exception messages in HQ are meant to be user-facing - I originally said "most" but that might not be true. Most view-level exceptions are user-facing, at least. The one in `link_domains` definitely looks useful for people to see.

The main ambiguous area I can think of is app manager, where there are some exceptions that are tailored for app builders but can also arise in web apps (where they won't make sense to a production user), so in those cases we might trap the exception and show a more generic error.

Reply:
Ha forgot that I changed the actual message associated with a DomainLinkError to be more user friendly, so I can likely revert this.